helix/machine_learning/learner.py

```diff
@@ -199,36 +236,73 @@ def _fit_kfold(self, data: Tuple) -> None:
                 res[i][model_name] = {}
                 self._logger.info(f"Fitting {model_name} for test fold sample {i+1}...")
                 model_type = get_model_type(model_name, self._problem_type)
-                model = model_type(**params["params"])
-                model.fit(X_train, y_train)
-                y_pred_train = model.predict(X_train)
-                res[i][model_name]["y_pred_train"] = y_pred_train
-                y_pred_test = model.predict(X_test)
-                res[i][model_name]["y_pred_test"] = y_pred_test
-                if self._problem_type == ProblemTypes.Classification:
-                    y_pred_probs_train = model.predict_proba(X_train)
-                    res[i][model_name]["y_pred_train_proba"] = y_pred_probs_train
-                    y_pred_probs_test = model.predict_proba(X_test)
-                    res[i][model_name]["y_pred_test_proba"] = y_pred_probs_test
+                model = get_model(model_type, params["params"])
+
+                # Handle MLREM learner differently
+                if hasattr(model, "fit") and model.__class__.__name__ == "MLREMLearner":
```

Comment:
Absolutely not. All models in Helix have to work the same way, following the scikit-learn API. We can't have special cases for every model that people want to add.